src/components/AttendanceSystem/AttendanceNoShowCharts.js

```diff
@@ -0,0 +1,295 @@
+              {selectedEvent.overallRating} / 5
+            </p>
+          </div>
+          <div>
+            <p style={{ fontSize: '14px', fontWeight: 'bold', marginBottom: '5px' }}>Status: </p>
+            <p style={{ fontSize: '20px', fontWeight: 'bold' }}>{selectedEvent.status}</p>
+          </div>
+          <div>
+            <p style={{ fontSize: '14px', fontWeight: 'bold', marginBottom: '5px' }}>
+              Total Registrations
+            </p>
+            <p style={{ fontSize: '20px', fontWeight: 'bold' }}>{selectedEvent.registrations}</p>
+          </div>
+          <div>
+            <p style={{ fontSize: '14px', fontWeight: 'bold', marginBottom: '5px' }}>
+              Total Attendees
+            </p>
+            <p style={{ fontSize: '20px', fontWeight: 'bold' }}>
+              {selectedEvent.attendees} (
+              {calculatePercentage(selectedEvent.attendees, selectedEvent.registrations)})
+            </p>
+          </div>
+          <div>
+            <p style={{ fontSize: '14px', fontWeight: 'bold', marginBottom: '5px' }}>Completed</p>
+            <p style={{ fontSize: '20px', fontWeight: 'bold' }}>
+              {selectedEvent.completed} (
+              {calculatePercentage(selectedEvent.completed, selectedEvent.attendees)})
+            </p>
+          </div>
+          <div>
+            <p style={{ fontSize: '14px', fontWeight: 'bold', marginBottom: '5px' }}>Walkouts</p>
+            <p style={{ fontSize: '20px', fontWeight: 'bold' }}>
+              {selectedEvent.walkouts} (
+              {calculatePercentage(selectedEvent.walkouts, selectedEvent.attendees)})
+            </p>
+          </div>
+        </div>
+      </div>
+      <div
+        style={{
+          backgroundColor: '#fff',
+          borderRadius: '8px',
+          boxShadow: '0 4px 6px rgba(0, 0, 0, 0.1)',
+          padding: '20px',
+          marginBottom: '20px',
+        }}
+      >
+        <h2 style={{ fontSize: '1.5em', marginBottom: '20px' }}>
+          Attendance and No-Show Breakdown
+        </h2>
+        <div
+          style={{
+            display: 'flex',
+            justifyContent: 'space-between',
+            flexWrap: 'nowrap',
+            gap: '20px',
+          }}
+        >
+          <div style={{ flexGrow: 1, height: '300px' }}>
+            <ResponsiveContainer width="100%" height="100%">
+              <PieChart>
+                <Pie
+                  data={attendanceData}
+                  cx="50%"
+                  cy="50%"
+                  labelLine={false}
+                  label={renderCustomizedLabel}
+                  outerRadius={80}
+                  fill="#8884d8"
+                  dataKey="value"
+                >
+                  {attendanceData.map(index => (
+                    <Cell
+                      key={`cell-${index}`}
+                      fill={attendanceColors[index % attendanceColors.length]}
+                    />
+                  ))}
```

Comment:
The problem is that you're using the actual value of the attendance data (index) as the key, but the index and not its position in the array.

```suggestion
                  {attendanceData.map((item, i) => (
                    <Cell
                      key={`cell-${i}`}
                      fill={attendanceColors[i % attendanceColors.length]}
                    />
                  ))}
```